pkg/controllers/plugin/pluginpreset_controller.go

```diff
@@ -218,6 +218,47 @@ func (r *PluginPresetReconciler) reconcilePluginPreset(ctx context.Context, pres
 	return skippedCondition, failedCondition, utilerrors.NewAggregate(allErrs)
 }
 
+func shouldSkipPlugin(plugin *greenhousev1alpha1.Plugin, preset *greenhousev1alpha1.PluginPreset) bool {
+	if plugin.Labels[greenhouseapis.LabelKeyPluginPreset] != preset.Name {
+		return true
+	}
+
+	if preset.Spec.Plugin.PluginDefinition != plugin.Spec.PluginDefinition {
```

Comment:
This is a valid thought, but we should deny changes in `Plugin.Spec.PluginDefinition` after Plugin `CREATE` in a webhook, wdyt @IvoGoman ?

Reply:
This validation is already in Place, so no need to check this here: https://github.com/cloudoperators/greenhouse/blob/ecba94afc093967434d260b75123a6bb7174bf59/pkg/admission/pluginpreset_webhook.go#L97-L99
But it seems this is not the case for the Plugin itself 👀 https://github.com/cloudoperators/greenhouse/blob/main/pkg/admission/plugin_webhook.go#L105

Reply:
I am taking care of updating the webhook, as there is some more cleanup to be done as well.

Reply:
Done